internal/app/repository/pg/pg.go

```diff
@@ -0,0 +1,316 @@
+	}
+
+
+	_, errWithdrawList := p.db.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS withdraw_list (
+    											id serial 		PRIMARY KEY,
+    											order_num	 	VARCHAR (255) NOT NULL,
+    											sum_points	 	FLOAT,
+    											user_id 		INT NOT NULL,
+                                   				processed_at 	TIMESTAMP default NOW()
+												)`)
+	if errWithdrawList != nil {
+		return fmt.Errorf("create table `withdraw_list`: %w", errWithdrawList)
+	}
+
+	return nil
+}
+
+// Ping - Проверка соединения с БД
+func (p *pg) Ping() bool {
+	if err := p.db.Ping(); err != nil {
+		log.Printf("Ping fail:, %s", err)
+		return false
+	}
+	return true
+}
+
+// GetUser вернет данные пользователя по логину
+
+// UserExist проверяет наличие пользователя в БД по логину
+func (p *pg) UserExist(ctx context.Context, login string) (bool, error) {
+	var loginFromDB string
+	err := p.db.QueryRowContext(ctx, `SELECT login FROM app_user WHERE login=$1 LIMIT 1`, login).Scan(&loginFromDB)
+	// Записи нет в БД
+	if fmt.Sprintf("%s", err) == "sql: no rows in result set" {
+		return false, nil
+	}
+	// Обрабатываем прочие ошибки
+	if err != nil {
+		return false, fmt.Errorf("SQL Query Error: %s", err)
+	}
+	// Запись есть но логин не совпадает
+	if login == loginFromDB {
+		return true, nil
+	}
+	// Default
+	return false, nil
+
+}
+
+// NewUser - создает нового пользователя и возвращает его id
+func (p *pg) NewUser(ctx context.Context, login string, hashPwd string) (int, error) {
+	lastInsertID := 0
+	err := p.db.QueryRow(`INSERT INTO app_user (login, password) VALUES ($1, $2) RETURNING id`, login, hashPwd).Scan(&lastInsertID)
+	if err != nil {
+		return 0, fmt.Errorf("new user insert error: %s", err)
+	}
+	fmt.Printf("lastInsertID %d for user %s\n", lastInsertID, login)
+	return lastInsertID, nil
+}
+
+// UserIsValid - Делает SQL в БД если по login и хеш пароля есть запись - значит пользователь существует и валиден.
+func (p *pg) UserIsValid(ctx context.Context, login string, hashPwd string) (userID int, err error) {
+	err = p.db.QueryRowContext(ctx, `SELECT app_user.id FROM app_user
+											WHERE login=$1
+											AND password=$2
+											LIMIT 1`, login, hashPwd).Scan(&userID)
+	if err != nil {
+		return 0, fmt.Errorf("user not found: %s", err)
+	}
+	return userID, nil
+}
+
+// GetOrderByNumber Вернет заказ по его номеру
+func (p *pg) GetOrderByNumber(ctx context.Context, orderNum string) (o OrderDB, err error) {
+	// Если записи нет то вернется {0 0 0  0 }
+	row := p.db.QueryRowContext(ctx, `SELECT id, user_id, number, status, accrual, uploaded_at 
+										   FROM app_order 
+										   WHERE number=$1 LIMIT 1`, orderNum)
+	err = row.Scan(&o.UserID, &o.UserID, &o.Number, &o.Status, &o.Accrual, &o.UploadedAt)
+	if err != nil {
+		return o, fmt.Errorf("order not found: %s", err)
+	}
+	return o, nil
+}
+
+// AddOrder добавит новый номер заказа
+func (p *pg) AddOrder(ctx context.Context, orderNum string, userID int) (err500 error) {
```

Comment:
Я бы предложил много где переехать на оперирование структурами, тогда если в один момент тебе понадобится прокидывать еще одно поле, достаточно будет добавить его в структуру, а так придется менять сигнатуру этого метода, а потом еще сигнатуру метода в интерфейсе
```suggestion
func (p *pg) AddOrder(ctx context.Context, order model.Order) (err error) {
```